extensions/wikia/CategoryPage3/CategoryPage3Hooks.class.php

```diff
@@ -0,0 +1,222 @@
+<?php
+
+class CategoryPage3Hooks {
+
+	const COOKIE_NAME = 'category-page-layout';
+	const GLOBAL_PREFERENCE_NAME = 'category-page-layout';
+
+	private static $oldQueryParams = [
+		'display',
+		'filefrom',
+		'fileuntil',
+		'page',
+		'pagefrom',
+		'pageuntil',
+		'sort',
+		'subcatfrom',
+		'subcatuntil',
+	];
+
+	/**
+	 * @param $categoryInserts
+	 * @param $categoryDeletes
+	 * @param $title
+	 * @return bool
+	 * @throws MWException
+	 */
+	public static function onAfterCategoriesUpdate( $categoryInserts, $categoryDeletes, $title ): bool {
+		$categories = $categoryInserts + $categoryDeletes;
+
+		foreach ( array_keys( $categories ) as $categoryTitle ) {
+			$title = Title::newFromText( $categoryTitle, NS_CATEGORY );
+
+			CategoryPage3CacheHelper::setTouched( $title );
+		}
+
+		return true;
+	}
+
+	/**
+	 * @param Title $title
+	 * @param Article $article
+	 * @return bool
+	 */
+	public static function onArticleFromTitle( $title, &$article ): bool {
+		if ( $title->isRedirect() ) {
+			$title = $article->getRedirectTarget();
+		}
+
+		if ( is_null( $title ) || !$title->inNamespace( NS_CATEGORY ) ) {
```

Comment:
this null check is redundant, since the method call above will cause an error if `$title` is null

Reply:
I think if there is a redirect but it's invalid, then it could be `null`.